#! /bin/sh -e

# DP: Use sensible-editor instead of vi as fallback editor

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

--- gcc/gccbug.in~	2003-03-01 00:51:42.000000000 +0100
+++ gcc/gccbug.in	2003-03-02 12:08:36.000000000 +0100
@@ -134,7 +134,7 @@
 # If they don't have a preferred editor set, then use
 if [ -z "$VISUAL" ]; then
   if [ -z "$EDITOR" ]; then
-    EDIT=vi
+    EDIT=/usr/bin/sensible-editor
   else
     EDIT="$EDITOR"
   fi
